Adding tests for Short Code Resolver

DillonAd opened the pull request:
I believe I covered the cases in #5, but if there is anything else you want to test let me know!

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
r is not a convenient name, should be something like redirectResult

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+        public async Task ResolveShortCode_Fail_CodeNotFound()
+        {
+            // Assemble
+            var targetEndpoint = "https://google.com";
```
This var is never used

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
Again, naming here

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
Again, naming here

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+
+            // Assert
+            Assert.IsInstanceOf<RedirectResult>(result);
+            var r = result as RedirectResult;
```
Again, naming here

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+        public async Task ResolveShortCode_Fail_EmptyEndpoint(string endpoint)
+        {
+            // Assemble
+            var targetEndpoint = "https://google.com";
```
This var is never used

nikneem commented on src/HexMaster.ShortLink.Tests/ResolveShortCodeFunction_Tests.cs:
```diff
@@ -0,0 +1,157 @@
+            
+            var collectorMock = new Mock<IAsyncCollector<LinkClickedMessage>>();
+
+            var entity = new ShortLinkEntity() { EndpointUrl = endpoint };
```
This one is never used

nikneem reviewed (commented):
I'm going to approve this PR after some minor tweaks, but I don't understand why you went for this level of complexity. You're now not testing the Resolver function, but the CORE, which has it's own unit tests.

But I really appreciate your work, was this top of mind knowledge or a lot of google?

DillonAd commented:
I fixed the naming and removed the unused variables. I also noticed that I forgot to rename the class so I fixed that as well. 

The overall idea was to not alter the existing code if at all possible since it is already functioning correctly. This does touch a couple elements of the framework, but I couldn't find another good way to get to the returned URL without making changes to the existing code.

I've been writing a lot of tests lately, but I did have to spend some time on Google to look up how to mock the `TableQuerySegment` class properly. :smiley:
